pkg/cluster/node.go

```diff
@@ -297,7 +297,13 @@ func patchServerSpec(node *k3d.Node) error {
 	node.Labels[k3d.LabelServerAPIHost] = node.ServerOpts.ExposeAPI.Host
 	node.Labels[k3d.LabelServerAPIPort] = node.ServerOpts.ExposeAPI.Port
 
-	node.Args = append(node.Args, "--tls-san", node.ServerOpts.ExposeAPI.Host) // add TLS SAN for non default host name
+	// If the runtime is docker, attempt to use the docker host
+	if runtime.ID() == "docker" && runtime.GetHost() != "" {
```

Comment:
```suggestion
	if runtime.GetHost() != "" {
```
If runtimes which don't support this just return an empty string, we can drop this runtime-specific check here, right?